dev/plugins/hu.elte.txtuml.xtxtuml.ui/src/hu/elte/txtuml/xtxtuml/ui/contentassist/XtxtUMLReferenceProposalCreator.xtend

```diff
@@ -274,16 +271,12 @@ class XtxtUMLReferenceProposalCreator extends XbaseReferenceProposalCreator {
 			// be difficult to achieve starting from a plain JvmType
 			proposedObj instanceof JvmDeclaredType && (proposedObj as JvmDeclaredType).superTypes.exists [
 				val typeRef = toLightweightTypeReference;
-				typeRef.isSubtypeOf(DataType) || typeRef.isInterface && typeRef.isSubtypeOf(ExternalType) || typeRef.isSubtypeOf(ModelEnum) ||
```

Comment:
The enclosing entity of the part of the code where we are about to propose or the enclosing entity of the proposal? I think the first one, but I have no idea how to implement this.